classes/ObjectModel.php

```diff
@@ -363,6 +363,24 @@ public function getFieldsLang()
         return $fields;
     }
 
+    /**
+     * Returns the language related to the object or the default one if it doesn't exists
+     *
+     * @return Language
+     *
+     * @throws PrestaShopDatabaseException
+     * @throws PrestaShopException
+     */
+    public function getAssociatedLanguage(): Language
+    {
+        $language = new Language($this->id_lang);
```

Comment:
I wonder if this should be stored in a field to avoid creating a new Object each time the function is called?

Reply:
It could be updated in cas id_lang is changed by checking:

```php
if (null === $this->affectedLanguage || (int) $this->affectedLanguage->getId() !== $this->id_lang)
```

Reply:
Oh but this check wouldn't work when the language has been removed since it will never match the default lang id